Index: Skylark/App.cpp
===================================================================
--- Skylark/App.cpp	(revision 8303)
+++ Skylark/App.cpp	(working copy)
@@ -173,7 +173,7 @@
 {
 //	DisableHUP();
 	if(static_dir.GetCount() == 0)
-		static_dir = root + "/static";
+		static_dir = root.IsEmpty() ? "static" : (root + "/static");
 
 	SqlSession::PerThread();
 	SqlId::UseQuotes();
Index: Skylark/Static.icpp
===================================================================
--- Skylark/Static.icpp	(revision 8303)
+++ Skylark/Static.icpp	(working copy)
@@ -15,7 +15,7 @@
 	
 	String path = GetFileOnPath(file, SkylarkApp::Config().path, false);
 	if(path.GetCount())
-		return SkylarkApp::Config().static_dir + "/ver" + GetFileTimeText(path) + "/" + file;
+		return "/" + SkylarkApp::Config().static_dir + "/ver" + GetFileTimeText(path) + "/" + file;
 	return Null;
 }
 
